server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCVectorizedMappingFunctions.java

```diff
@@ -511,6 +615,76 @@ private static TriConsumer<ColumnVector, Integer, Object> getTimestampInLocalWri
+     *                                will be the underlying primitive type. This category is used to determine the write
+     *                                function that will be used to write the primitive value in the child column vector
+     * @return a function setting the list column vector
+     */
+    private static TriConsumer<ColumnVector, Integer, Object> getListWriteFunction(TypeDescription.Category underlyingChildCategory) {
+        return getListWriteFunction(underlyingChildCategory, writeFunctionsMap.get(underlyingChildCategory));
+    }
+
+    /**
+     * A special function that writes lists to ORC file
+     * @param underlyingChildCategory the underlying primitive child category. As PXF currently only has the capability to infer
+     *                                an ORC schema from the GPDB schema, so all arrays will be one-dimensional, and this value
+     *                                will be the underlying primitive type
+     * @param childWriteFunction      the write function that will be used to write the primitive value in the child column vector
+     * @return a function setting the list column vector
+     */
+    private static TriConsumer<ColumnVector, Integer, Object> getListWriteFunction(TypeDescription.Category underlyingChildCategory, final TriConsumer<ColumnVector, Integer, Object> childWriteFunction) {
+        return (columnVector, row, val) -> {
+            // TODO: as all schemas right now are auto-generated, the columnVector will always be a ListColumnVector
+            // when we allow user-generated schemas, do we need to consider checking the type of the columnvector before casting?
+            ListColumnVector listColumnVector = (ListColumnVector) columnVector;
+
+            List<Object> data = orcUtilities.parsePostgresArray(val.toString(), underlyingChildCategory);
+
+            int length = data.size();
+            // the childCount refers to "the number of children slots used"
+            int offset = listColumnVector.childCount;
+
+            // set the offset, length and new childCount for this row
+            listColumnVector.offsets[row] = offset;
+            listColumnVector.lengths[row] = length;
+            listColumnVector.childCount += length;
+            // if the number of children slots needed is greater than the current size
+            if (listColumnVector.childCount > listColumnVector.child.isNull.length) {
+                // reallocate to ensure that the columnvector can hold the data
+                // use the average row size to calculate what the new size should be
+                // (childCount/row) * (batchsize - row) => assume average size and # of rows
+                float avgRow = listColumnVector.childCount / (row+1);
+                listColumnVector.child.ensureSize(Math.round(avgRow) * (VectorizedRowBatch.DEFAULT_SIZE - (row+1)),   true);
```

Comment:
shouldn't we give the overall new size required, not the increment ?
```suggestion
                listColumnVector.child.ensureSize(oldChildSize + Math.ceil(avgRow) * (listColumnVector.length - (row+1)),   true);
```